middleware/useContext.js

```diff
@@ -0,0 +1,15 @@
+require("express-async-errors");
+const jwt = require("jsonwebtoken");
+const { User } = require("../models/user");
+const validateToken = require("./validateToken");
+
+module.exports = async (req, res) => {
+  const token = validateToken(req, res);
+
+  const decoded = jwt.verify(token, process.env.JWT_PRIVATE_KEY);
+
+  const user = await User.findOne({ _id: decoded._id });
+  if (!user) return res.status(400).send(`Invalid request !`);
+
+  return decoded;
```

Comment:
I would suggest to have context as a separate function in another directory in order to access the context value. Also, the naming could suffice by being just `context`. You would need to cal it, so there is no need to be an anonymous function.